docs/source/search.mdx

```diff
@@ -0,0 +1,153 @@
+# Search text in a dataset
+
+Datasets Server provides a `/search` endpoint for searching words in a dataset.
+
+<Tip warning={true}>
+  Currently, only {" "}
+  <a href="./parquet">datasets with parquet exports</a>
+  are supported so Datasets Server can index the contents and run the search without downloading the
+  whole dataset.
+</Tip>
+
+This guide shows you how to use Datasets Server's `/search` endpoint to search for a query string.
+Feel free to also try it out with [ReDoc](https://redocly.github.io/redoc/?url=https://datasets-server.huggingface.co/openapi.json#operation/searchRows).
+
+The text is searched in the columns of type `string` or `ClassLabel`, even if the values are nested in a dictionary.
+
+The `/search` endpoint accepts five query parameters:
+
+- `dataset`: the dataset name, for example `glue` or `mozilla-foundation/common_voice_10_0`
+- `config`: the configuration name, for example `cola`
+- `split`: the split name, for example `train`
+- `query`: the text to search
+- `offset`: the offset of the slice, for example `150`
+- `length`: the length of the slice, for example `10` (maximum: `100`)
+
```

Comment:
I think here it would be nice to explicitly say what we're searching for here (can remove the mention of this later in the doc). Something like:

"For example, let's search for the text `dog` in the `train` split of the `SelfRC` configuration of the `duorc` dataset from slice 150-151:"